dist/tools/format-board-list/format-board-list.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+import os
+import sys
+import subprocess
+import re
+from pprint import pprint
+from tempfile import NamedTemporaryFile
+import shutil
+
+TARGET_VARIABLES = ['BOARD_INSUFFICIENT_MEMORY', 'BOARD_BLACKLIST', 'BOARD_WHITELIST']
+
+def chdir_to_riot_root():
+	SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+	RIOT_DIR = os.path.realpath(SCRIPT_DIR + "/../../..")
```

Comment:
I'd prefer a script that does not need to live in a specific location in order to work properly.

Reply:
The thought behind this is that I the script is working-directory agnostic. I'm happy with adding an argument that specifies the target directory if that seems useful to anyone, but the default mode of operation, in my opinion, should be "just run it and it will do the correct thing". Given that the script is pretty quick, it's also not a problem to always run it over the entirety of the RIOT code base.

Reply:
To expand a bit more: in the case of a general-purpose script, i.e. one that is not intended to live in a particular directory in the RIOT source tree, I'd agree with you. In this case, however, I _personally_ value "it does the right thing automatically" higher than the potential flexibility of being able to pass the target directory.

Furthermore, being able to pass the target directory would make the script potentially more complicated with questions such as "does it also have to work in a directory that is not part of a git repository" or, if the answer is no, checking if I'm in a git repository, ...